### PR TITLE
Clean up dependencies, move to dev/peer correspondingly

### LINKED ISSUE #33: Clean up dependencies
Current dependencies list include a few things not suitable there, like react, react-dom, react-scripts. They should be moved to dev dep and react should be peer too.
/assign @kwasi

### PR DESCRIPTION
**Pull request by Bobgy**
/assign @kwasi 
/assign @avdaredevil 

Fixes https://github.com/kubeflow/frontend/issues/33
The change helps with:
1. the massive dependencies from react-scripts won't be installed by dependants of @kubeflow/frontend
2. libraries that should better be using the same version are added to both peer and dev dependencies to guarantee so.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/frontend/34)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,9 +72,18 @@
     "enzyme": "^3.10.0",
     "enzyme-adapter-react-16": "^1.14.0",
     "enzyme-to-json": "^3.3.5",
-    "typescript": "^3.7.4"
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
+    "react-router-dom": "^4.3.1",
+    "react-scripts": "^3.3.1",
+    "typescript": "^3.7.4",
+    "typestyle": "^2.0.4"
```
Comment (avdaredevil) [inline package.json:80]: I don't think these fall under `devDepencies`, as they are run-time dependencies in production.

Comment (avdaredevil) [inline package.json:80]: Does it make sense to include `typestyle` and similar in both `devDependencies` and `peerDependencies`?

**Review by avdaredevil — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Bobgy) [reply package.json:80]: They are added as peer to make sure they will be installed by dependants of this package. This guarantees both packages are using the same version.

Comment (Bobgy) [reply package.json:80]: yes, they are included as both

Comment (avdaredevil) [reply package.json:80]: I see. Thanks for clarifying that!

Comment (avdaredevil) [reply package.json:80]: Great!

**Pull request merged**